### PR TITLE
Update tests for TypedArrays foreach and map

### PR DESCRIPTION
**Pull request by leobalter**
Based on the feedback from gh-593

### CODE REVIEW

File: test/built-ins/TypedArray/prototype/forEach/values-are-not-cached.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  Integer indexed values are not cached before interaction
+info: >
+  22.2.3.12 %TypedArray%.prototype.forEach ( callbackfn [ , thisArg ] )
+
+  %TypedArray%.prototype.forEach is a distinct function that implements the same
+  algorithm as Array.prototype.forEach as defined in 22.1.3.10 except that the
+  this object's [[ArrayLength]] internal slot is accessed in place of performing
+  a [[Get]] of "length"
+includes: [testTypedArray.js]
+features: [Reflect.set]
```
Comment (leobalter) [inline test/built-ins/TypedArray/prototype/forEach/values-are-not-cached.js]: removing...

File: test/built-ins/TypedArray/prototype/forEach/values-are-not-cached.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  %TypedArray%.prototype.forEach is a distinct function that implements the same
+  algorithm as Array.prototype.forEach as defined in 22.1.3.10 except that the
+  this object's [[ArrayLength]] internal slot is accessed in place of performing
+  a [[Get]] of "length"
+includes: [testTypedArray.js]
+---*/
+
+testWithTypedArrayConstructors(function(TA) {
+  var sample = new TA([42, 43, 44]);
+  var calls = 0;
```
Comment (goyakin) [inline test/built-ins/TypedArray/prototype/forEach/values-are-not-cached.js]: I don't think this was intended to be used in this test. If so, can we remove it?

File: test/built-ins/TypedArray/prototype/map/values-are-not-cached.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+description: >
+  Integer indexed values changed during iteration
+info: >
+  22.2.3.19 %TypedArray%.prototype.map ( callbackfn [ , thisArg ] )
+includes: [testTypedArray.js]
+---*/
+
+testWithTypedArrayConstructors(function(TA) {
+  var sample = new TA([42, 43, 44]);
+  var calls = 0;
```
Comment (goyakin) [inline test/built-ins/TypedArray/prototype/map/values-are-not-cached.js]: We should remove this one too if we don't intend to use it.

**Comment by goyakin** [comment]
@leobalter Other than my comments for the `calls` variables, everything else LGTM.

**Comment by jugglinmike** [comment]
I agree with Gorkem; this looks good to merge once those variables are removed.

**Comment by leobalter** [comment]
fixes are up! Thanks!

**Comment by goyakin** [comment]
Merged as 84d9507.

**Pull request closed without merging**